src/EditorFeatures/Core/Implementation/EditAndContinue/EditAndContinueSaveFileCommandHandler.cs

```diff
@@ -36,14 +38,13 @@ void IChainedCommandHandler<SaveCommandArgs>.ExecuteCommand(SaveCommandArgs args
 
             if (Workspace.TryGetWorkspace(textContainer, out var workspace))
             {
-                var encService = workspace.Services.GetService<IEditAndContinueWorkspaceService>();
-                if (encService != null)
+                var documentId = workspace.GetDocumentIdInCurrentContext(textContainer);
+                if (documentId != null)
```

Comment:
this feels like this would indicate a bug, no?   should we assert/nfw here?

Reply:
if docId can legit be null, we should then doc the case here.

Reply:
Not sure. I'd rather not change this in this PR.